product_supplierinfo_barcode/views/product_views.xml

```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="product_template_search_view_supplier_code" model="ir.ui.view">
+        <field name="name">product_template_search_view_supplier_code</field>
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_search_view" />
+        <field name="arch" type="xml">
+            <field name="categ_id" position="after">
+                <field
+                    name="seller_ids"
+                    string="Supplier code"
+                    filter_domain="['|', '|', ('seller_ids.barcode', 'ilike', self), ('seller_ids.product_code', 'ilike', self), ('seller_ids.product_name', 'ilike', self)]"
+                />
+            </field>
+        </field>
+    </record>
```

Comment:
Why not migrate the view that adds the barcode to the supllierinfo tree in the product form?
Like in version 15: https://github.com/OCA/stock-logistics-barcode/blob/147a3020ebb92fb974f42e115fcc0bbba3b65188/product_supplierinfo_barcode/views/product_views.xml#L18-L27
Tested in local environment and works correctly.